### PR TITLE
Update Optimize FID guide

### PR DESCRIPTION
**Pull request by housseindjirdeh**
Changes proposed in this pull request:

- Updates "Optimize First Input Delay" guide:
  - Removes the [Minify and compress JavaScript files](https://web.dev/optimize-fid/#minify-and-compress-javascript-files) section
  - Moves the entire [Reduce JavaScript](https://web.dev/optimize-fid/#reduce-javascript-execution) section to the end of the article

Note: It may look like a lot of text was changed, but it's mostly sections being re-arranged. Going through each of the commits will show the changes easier.

### CODE REVIEW

**Comment by housseindjirdeh** [comment]
CC @anniesullie

File: src/site/content/en/blog/optimize-fid/index.md [hunk 1]
```diff
@@ -56,51 +56,107 @@ compiles, and executes on your web page will directly reduce FID.
...
-In addition to minimizing JavaScript, [breaking up long-running code into smaller tasks](#long-tasks) can allow the
-browser to run input handlers sooner, which can improve FID.
+FID should improve noticeably as you adopt best practices like code-splitting and breaking up your
+Long Tasks. While TBT is not a field metric, it's useful for checking progress towards ultimately
+improving both Time To Interactive (TTI) and FID.
 
-### Minify and compress JavaScript files
+{% Aside %}
+For more information, take a look at [Are long JavaScript tasks delaying your Time to
+Interactive?](/long-tasks-devtools/) guide goes into more detail about this topic.
```
Comment (kaycebasques) [inline src/site/content/en/blog/optimize-fid/index.md:86]:
```suggestion
Interactive?](/long-tasks-devtools/).
```

File: src/site/content/en/blog/optimize-fid/index.md [hunk 2]
```diff
@@ -56,51 +56,107 @@ compiles, and executes on your web page will directly reduce FID.
...
-    [rollup-plugin-terser](https://www.npmjs.com/package/rollup-plugin-terser) as an output plugin
+### First-party script execution can delay interaction readiness
 
-In addition to minification, compress your JavaScript resources to minimize their delivery size. If
-possible, use
-[Brotli](https://opensource.googleblog.com/2015/09/introducing-brotli-new-compression.html) which
-provides better compression results than
-[Gzip](https://www.youtube.com/watch?v=whGwm0Lky2s&feature=youtu.be&t=14m11s) and [can be used in
-almost all newer browsers](https://caniuse.com/#feat=brotli).
++   JS size bloat, heavy execution times and inefficient chunking can slow down how soon a
```
Comment (kaycebasques) [inline src/site/content/en/blog/optimize-fid/index.md:96]:
```suggestion
+   JavaScript size bloat, heavy execution times and inefficient chunking can slow down how soon a
```

File: src/site/content/en/blog/optimize-fid/index.md [hunk 3]
```diff
@@ -56,51 +56,107 @@ compiles, and executes on your web page will directly reduce FID.
...
+    page can respond to user input and impact FID, TBT and TTI. Progressive loading of code and
+    features can help spread this work out and improve interaction readiness.
++   Server-side rendered apps may look like they're getting pixels painted on the screen
+    quickly, but beware of user interactions being blocked by large script executions (e.g
```
Comment (kaycebasques) [inline src/site/content/en/blog/optimize-fid/index.md:100]:
```suggestion
    quickly, but beware of user interactions being blocked by large script executions (e.g.
```

File: src/site/content/en/blog/optimize-fid/index.md [hunk 4]
```diff
@@ -56,51 +56,107 @@ compiles, and executes on your web page will directly reduce FID.
...
+### First-party script execution can delay interaction readiness
 
-In addition to minification, compress your JavaScript resources to minimize their delivery size. If
-possible, use
-[Brotli](https://opensource.googleblog.com/2015/09/introducing-brotli-new-compression.html) which
-provides better compression results than
-[Gzip](https://www.youtube.com/watch?v=whGwm0Lky2s&feature=youtu.be&t=14m11s) and [can be used in
-almost all newer browsers](https://caniuse.com/#feat=brotli).
++   JS size bloat, heavy execution times and inefficient chunking can slow down how soon a
+    page can respond to user input and impact FID, TBT and TTI. Progressive loading of code and
```
Comment (kaycebasques) [inline src/site/content/en/blog/optimize-fid/index.md:97]:
```suggestion
    page can respond to user input and impact FID, TBT, and TTI. Progressive loading of code and
```

File: src/site/content/en/blog/optimize-fid/index.md [hunk 5]
```diff
@@ -56,51 +56,107 @@ compiles, and executes on your web page will directly reduce FID.
...
+Below are the TBT scores before and after optimizing first-party script loading for an
+application. By moving costly script loading (and execution) for a non-essential component off the
+critical path, users were able to interact with the page much sooner.
+
+<img class="w-screenshot" src="tbt-before-after-first-party.png" 
+     alt="Improvements in TBT score in Lighthouse after optimizing the first-party script.">
+
+### Data-fetching can impact many aspects of interaction readiness
+
++   Waiting on a waterfall of cascading fetches (e.g. JS and data fetches for components) can
```
Comment (kaycebasques) [inline src/site/content/en/blog/optimize-fid/index.md:114]:
```suggestion
+   Waiting on a waterfall of cascading fetches (e.g. JavaScript and data fetches for components) can
```

**Pull request merged**